### ISSUE TITLE
Feature Request: Ignore a subset of consul checks when verifying a deployment

### ISSUE THREAD

**Issue by margueritepd** [hashicorp/nomad#4252]
Hello!

I am trying to run a cluster on Nomad in which there is one master instance and multiple standbys. Only the master responds to requests, and the standbys do not forward requests to the master.

I want to add two Consul checks to my nomad job:
* A health check to verify that the deployed instances are healthy
* Another check to advertise which is the leader, so that clients can discover the leader and send requests to the right place

I want the first health check to be used to verify that a deployment is healthy, but I _don't_ want to use the second one for that same purpose since it would mark the standbys as unhealthy.

I have found a workaround that allows me to do this, but I am not confident that it _should_ allow me to do this. Namely, in the definition of the check for "which one is the leader", I have the following:

```
check_restart {
  # This _may_ be undefined behaviour, but the intended (and actual) effect of using a
  # limit of 0 is that the health check failure won't halt deployment or cause restarts.
  limit = 0
  grace = "10s"
}
```

Can you confirm that this is an intended use of `check_restart`? If this _is_ an intended use, it might be worth documenting somewhere. If it _isn't_, I want to submit a feature request for an officially supported way of doing what I'm trying to do.

Thanks!

**Comment by schmichael** [comment]
`check_restart` itself does not affect deployments. You would have to use `task_state` based deployments instead of health check based deployments to get this setup to work currently.

`limit=0` is the same as omitting the `check_restart` stanza entirely. Omitting the `check_restart` stanza for the second check will prevent Nomad from restarting non-leaders. The check/service for non-leaders will be marked as failing which should accomplish your goals.

Allowing jobs to include/exclude checks for consideration during deployments seems like a reasonable feature to me.

**Comment by margueritepd** [comment]
It sounds like if my workaround _does_ work, it's unexpected and I shouldn't rely on that behaviour. Thanks for your response and I'll keep this ticket open as a feature request :)

**Comment by tgross** [comment]
@schmichael @drewbailey I bumped into this issue while searching for something else... should this be resolved by the liveness/readiness feature that shipped in Nomad 1.1.0?

**Comment by drewbailey** [comment]
@tgross yes 👍 linking docs: https://www.nomadproject.io/docs/job-specification/service#on_update
